knotx-core/src/main/java/io/knotx/launcher/KnotxCommand.java

```diff
@@ -0,0 +1,215 @@
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Scanner;
+
+@Name("run-knotx")
+@Summary("Runs a Knot.x instance")
+public class KnotxCommand extends BareCommand {
+
+  /**
+   * Error code used when the knotx configuration is missing or it's empty
+   */
+  private static final int KNOTX_MISSING_OR_EMPTY_CONFIGURATION_EXIT_CODE = 30;
+
+  private static final String DEFAULT_LOADER_FILE = "bootstrap.json";
+
+
+  private DeploymentOptions deploymentOptions;
+
+  private String config;
+
+  private boolean ha;
+  private boolean cluster;
+
+  private static final String KNOTX_STARTER_VERTICLE = KnotxStarterVerticle.class.getName();
+
+
+  /**
+   * Enables / disables the high-availability.
+   *
+   * @param ha whether or not to enable the HA.
+   */
+  @Option(longName = "ha", acceptValue = false, flag = true)
+  @Description(
+      "If specified the Knot.x will be deployed as a high availability (HA) deployment. This means it can "
+          +
+          "fail over to any other nodes in the cluster started with the same HA group.")
+  public void setHighAvailability(boolean ha) {
+    this.ha = ha;
+  }
+
+  /**
+   * Enables / disables the clustering.
+   *
+   * @param cluster whether or not to start Knot.x in clustered mode.
+   */
+  @Option(longName = "cluster", acceptValue = false, flag = true)
+  @Description(
+      "If specified then the Knot.x instance will form a cluster with any other Knotx.x instances on the "
+          +
+          "network.")
+  public void setCluster(boolean cluster) {
+    this.cluster = cluster;
+  }
+
+  /**
+   * The path to the Knot.x configuration loader json file
+   *
+   * @param path the path to the confing loader json
+   */
+  @Option(longName = "conf", argName = "config")
+  @Description("Specifies path to the config loader json Knot.x requires to start. " +
+      "If not specified a '" + DEFAULT_LOADER_FILE + "' in the class path is to be used.")
+  public void setConfig(String path) {
+    if (path != null) {
+      this.config = path;
+    } else {
+      this.config = null;
+    }
+  }
+
+  /**
+   * Validates the command line parameters.
+   *
+   * @param context - the execution context
+   * @throws CLIException - validation failed
+   */
+  @Override
+  public void setUp(ExecutionContext context) throws CLIException {
+    super.setUp(context);
+  }
+
+  /**
+   * @return whether the {@code cluster} option or the {@code ha} option are enabled.
+   */
+  @Override
+  public boolean isClustered() {
+    return cluster || ha;
+  }
+
+  @Override
+  public boolean getHA() {
+    return ha;
+  }
+
+  /**
+   * Starts vert.x and deploy the verticle.
+   */
+  @Override
+  public void run() {
+    JsonObject conf = getConfiguration();
+    if (conf == null) {
+      ExecUtils.exit(KNOTX_MISSING_OR_EMPTY_CONFIGURATION_EXIT_CODE);
+    }
+
+    super.run();
```

Comment:
Not sure if it is the correct solution. Please check `io.vertx.core.impl.launcher.commands.BareCommand#afterStartingVertx` method which is called when vert.x instance is set - those hooks should happen after Knot.x verticles deployment. In the current solution those hooks happen before event Knot.x Starter Verticle run.